src/test/integration/tests/basic.ts

```diff
@@ -12,232 +12,234 @@
  * http://polymer.github.io/PATENTS.txt
  */
 
-import {html} from 'lit-html';
+import {html, noChange, nothing} from 'lit-html';
 import {repeat} from 'lit-html/directives/repeat.js';
 
 import { SSRTest } from './ssr-test';
 
+const filterNodes = (nodes: ArrayLike<Node>, nodeType: number) =>
+  Array.from(nodes).filter(n => n.nodeType === nodeType);
+
+const testSymbol = Symbol();
+const testObject = {};
+const testArray = [1,2,3];
+const testFunction = () => 'test function';
+
 export const tests: {[name: string] : SSRTest} = {
 
-  'textExpression': {
+  'NodePart accepts a string': {
     render(x: any) {
       return html`<div>${x}</div>`;
     },
     expectations: [
       {
-        args: ['TEST'],
-        html: '<div>TEST</div>'
+        args: ['foo'],
+        html: '<div>foo</div>'
       },
       {
-        args: ['TEST2'],
-        html: '<div>TEST2</div>'
+        args: ['foo2'],
+        html: '<div>foo2</div>'
       }
     ],
     stableSelectors: ['div'],
   },
 
+  'NodePart accepts a number': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
+    },
+    expectations: [
+      {
+        args: [123],
+        html: '<div>123</div>'
+      },
+      {
+        args: [456.789],
+        html: '<div>456.789</div>'
+      }
+    ],
+    stableSelectors: ['div'],
+  },
 
-  'twoTextExpression': {
-    render(x: any, y: any) {
-      return html`<div>${x}${y}</div>`;
+  'NodePart accepts undefined': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
     },
     expectations: [
       {
-        args: ['A', 'B'],
-        html: '<div>A\n  B</div>'
+        args: [undefined],
+        html: '<div></div>'
       },
       {
-        args: ['C', 'D'],
-        html: '<div>C\n  D</div>'
+        args: ['foo'],
+        html: '<div>foo</div>'
       }
     ],
     stableSelectors: ['div'],
   },
 
-  'nested templates': {
-    render(x: any, y: any) {
-      return html`<div>${x}${html`<span>${y}</span>`}</div>`;
+  'NodePart accepts null': {
+    render(x: any) {
+      return html`<div>${x}</div>`;
     },
     expectations: [
       {
-        args: ['A', 'B'],
-        html: '<div>A\n  <span>B</span></div>'
+        args: [null],
+        html: '<div></div>'
       },
       {
-        args: ['C', 'D'],
-        html: '<div>C\n  <span>D</span></div>'
+        args: ['foo'],
+        html: '<div>foo</div>'
       }
     ],
-    stableSelectors: ['div', 'span'],
+    stableSelectors: ['div'],
   },
 
-  'attributeExpression': {
+  'NodePart accepts noChange': {
```

Comment:
Now that we have this test, the next one I'm very interested in is rendering on the server with a value, then hydrating with `noChange`. We'll need to update the test runner to allow this.